src/workers/deleteUser.ts

```diff
@@ -35,63 +37,67 @@ interface UserData {
-    const data: UserData = messageToJson(message);
-    try {
-      await con.transaction(async (entityManager): Promise<void> => {
-        await entityManager.getRepository(View).delete({ userId: data.id });
-        await entityManager.getRepository(Alerts).delete({ userId: data.id });
-        await entityManager
-          .getRepository(BookmarkList)
-          .delete({ userId: data.id });
-        await entityManager.getRepository(Bookmark).delete({ userId: data.id });
-        await entityManager
-          .getRepository(CommentUpvote)
-          .delete({ userId: data.id });
-        await entityManager.getRepository(Comment).delete({ userId: data.id });
-        await entityManager.getRepository(DevCard).delete({ userId: data.id });
-        await entityManager.getRepository(Feed).delete({ userId: data.id });
-        await entityManager
-          .getRepository(HiddenPost)
-          .delete({ userId: data.id });
-        await entityManager
-          .getRepository(PostReport)
-          .delete({ userId: data.id });
-        await entityManager.getRepository(Settings).delete({ userId: data.id });
-        await entityManager
-          .getRepository(SourceDisplay)
-          .delete({ userId: data.id });
-        await entityManager
-          .getRepository(SourceRequest)
-          .delete({ userId: data.id });
-        await entityManager.getRepository(Upvote).delete({ userId: data.id });
-        await entityManager
-          .getRepository(Post)
-          .update({ authorId: data.id }, { authorId: null });
-        await entityManager.getRepository(User).delete(data.id);
-      });
+export const deleteUser = async (
+  con: Connection,
+  logger: FastifyLoggerInstance,
+  userId: string,
+  messageId?: string,
+) => {
+  try {
+    await con.transaction(async (entityManager): Promise<void> => {
+      await entityManager.getRepository(View).delete({ userId });
+      await entityManager.getRepository(Alerts).delete({ userId });
+      await entityManager.getRepository(BookmarkList).delete({ userId });
+      await entityManager.getRepository(Bookmark).delete({ userId });
+      await entityManager.getRepository(CommentUpvote).delete({ userId });
+      await entityManager.getRepository(Comment).delete({ userId });
+      await entityManager.getRepository(DevCard).delete({ userId });
+      await entityManager.getRepository(Feed).delete({ userId });
+      await entityManager.getRepository(HiddenPost).delete({ userId });
+      await entityManager.getRepository(PostReport).delete({ userId });
+      await entityManager.getRepository(Settings).delete({ userId });
+      await entityManager.getRepository(SourceDisplay).delete({ userId });
+      await entityManager.getRepository(SourceRequest).delete({ userId });
+      await entityManager.getRepository(Upvote).delete({ userId });
+      await entityManager
+        .getRepository(Post)
+        .update({ authorId: userId }, { authorId: null });
```

Comment:
We need to another statement for scoutId

Reply:
Oh dang, well spotted! 🤯